### PR TITLE
[3.0] Upgrade kernel to 6.1.58.1

### PR DESCRIPTION
**Pull request by rlmenge**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Initial upgrade to kernel 6.1.58.1. 
Remove the imx8 dtb subpackage
Add patch to build systemd with new kernel-headers. This can be removed when systemd is updated https://github.com/systemd/systemd/commit/bbe53713455be38c0a587626439fd171f28c77fc 
Update strace to match kernel version
 
Use default config options for new configs. A quick config analysis can be seen below. Note that the DEBUGINFO option for aarch64 has been turned off but the debuginfo package is still being generated. This will be included in investigations for kernel configs in subsequent PRs.

The following will happen in follow-up PRs 
- update the specialised kernels
- trim the no longer needed kernel CVEs
- remove the nvme patch
- Comb config changes for the most optimized performance and function (defaults for now)

###### Change Log  

- Initial upgrade to kernel 6.1.58.1. 

###### Does this affect the toolchain?  

**YES**

###### Associated issues  

-  https://microsoft.visualstudio.com/OS/_dashboards/dashboard/990a43a2-4428-4ee2-8205-36aadc55d552

###### Test Methodology

image

**#########Pipeline Runs#########**

ARM:
toolchain: [Pipelines - Run 2.0.20231103-rlmenge-445346 (azure.com)](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=445360&view=results)
RPMS: [Pipelines - Run 2.0.20231103-rlmenge-445346 (azure.com)](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=445692&view=results)
image: [Pipelines - Run 2.0.20231103-rlmenge-445346 (azure.com)](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=445991&view=results)

AMD
toolchain: [Pipelines - Run 2.0.20231103-rlmenge-445346 (azure.com)](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=445360&view=results)
RPMS: [Pipelines - Run 20231106.1 (azure.com)](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=447148&view=results) 
image: [Pipelines - Run 2.0.20231103-rlmenge-445346 (azure.com)](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=447757&view=results)

image

**#########Config change overview#########**

This PR upgrades the 5.15.133.1 kernel to 6.1. Here is the rbrief list of configs which had values changed (does not include new or removed configs)

**AMD**
Positional argument 1: config-5.15.133.1
Positional argument 2: ../CBL-Mariner-main/SPECS/kernel/config
Optional argument: None
########### Common options with different values [config-5.15.133.1 : ../CBL-Mariner-main/SPECS/kernel/config] ###########
CONFIG_DRM_TTM_HELPER = m : y
CONFIG_DRM_GEM_SHMEM_HELPER = y : m
CONFIG_RADIO_ADAPTERS = y : m
CONFIG_CRC64 = m : y
CONFIG_ND_BTT = y : m
CONFIG_MEDIA_CONTROLLER = is not set : y

**ARM**
Positional argument 1: config_aarch64-5.15.133.1
Positional argument 2: ../CBL-Mariner-main/SPECS/kernel/config_aarch64
Optional argument: None
########### Common options with different values [config_aarch64-5.15.133.1 : ../CBL-Mariner-main/SPECS/kernel/config_aarch64] ###########
CONFIG_MEDIA_CONTROLLER = is not set : y
CONFIG_PINCTRL_OCELOT = y : is not set
CONFIG_CAN_UCAN = m : is not set
CONFIG_IRQ_BYPASS_MANAGER = y : m
CONFIG_CAN_M_CAN = m : is not set
CONFIG_CAN_CC770 = m : is not set
CONFIG_RADIO_ADAPTERS = y : m
CONFIG_FB_SYS_FOPS = y : m
CONFIG_SCSI_UFSHCD_PLATFORM = m : is not set
CONFIG_COMPAT_BRK = is not set : y
CONFIG_CRYPTO_SHA1_ARM64_CE = m : is not set
CONFIG_CRYPTO_SM3_ARM64_CE = m : is not set
CONFIG_CRYPTO_GHASH_ARM64_CE = m : is not set
CONFIG_CRYPTO_RNG_DEFAULT = y : m
CONFIG_CRYPTO_DES = m : is not set
CONFIG_CRYPTO_AES_ARM64_NEON_BLK = m : is not set
CONFIG_CAN_GS_USB = m : is not set
CONFIG_CRYPTO_SM4_ARM64_CE = m : is not set
CONFIG_CAN_IFI_CANFD = m : is not set
CONFIG_USERFAULTFD = y : is not set
CONFIG_ZSWAP = y : is not set
CONFIG_DRM_GEM_SHMEM_HELPER = y : m
CONFIG_CAN_8DEV_USB = m : is not set
CONFIG_SCSI_UFSHCD_PCI = m : is not set
CONFIG_CRYPTO_SHA2_ARM64_CE = m : is not set
CONFIG_CAN_MCBA_USB = m : is not set
CONFIG_DRM_KMS_HELPER = y : m
CONFIG_CAN_MCP251X = m : is not set
CONFIG_SLAB_FREELIST_HARDENED = y : is not set
CONFIG_CAN_GRCAN = m : is not set
CONFIG_CAN_EMS_USB = m : is not set
CONFIG_CAN_SOFTING = m : is not set
CONFIG_CAN_XILINXCAN = m : is not set
CONFIG_CRYPTO_AES_ARM64 = m : is not set
CONFIG_CRYPTO_AES_ARM64_BS = m : is not set
CONFIG_FB_SYS_COPYAREA = y : m
CONFIG_CRYPTO_AES_ARM64_CE_BLK = m : is not set
CONFIG_CAN_C_CAN = m : is not set
CONFIG_CAN_FLEXCAN = m : is not set
CONFIG_CAN_PEAK_PCIEFD = m : is not set
CONFIG_CRYPTO_AEAD = y : m
CONFIG_PINCTRL_MT8195 = is not set : y
CONFIG_ND_BTT = y : m
CONFIG_CAN_VCAN = m : is not set
CONFIG_CRYPTO_SHA512_ARM64 = m : is not set
CONFIG_DEBUG_INFO_DWARF_TOOLCHAIN_DEFAULT = y : is not set
CONFIG_CEC_CORE = y : m
CONFIG_CRYPTO_AES_ARM64_CE_CCM = m : is not set
CONFIG_CRYPTO_SHA512_ARM64_CE = m : is not set
CONFIG_SLAB_FREELIST_RANDOM = y : is not set
CONFIG_CAN_RCAR_CANFD = m : is not set
CONFIG_CRYPTO_ARC4 = m : is not set
CONFIG_CRYPTO_AES_ARM64_CE = m : is not set
CONFIG_CRYPTO_SEQIV = y : m
CONFIG_CAN_VXCAN = m : is not set
CONFIG_CAN_SLCAN = m : is not set
CONFIG_CAN_HI311X = m : is not set
CONFIG_CAN_SJA1000 = m : is not set
CONFIG_CAN_RCAR = m : is not set
CONFIG_CRYPTO_SHA256_ARM64 = m : is not set
CONFIG_CAN_PEAK_USB = m : is not set
CONFIG_CRC64 = m : y
CONFIG_ZSMALLOC = y : m
CONFIG_SLAB_MERGE_DEFAULT = is not set : y
CONFIG_FB_SYS_FILLRECT = y : m
CONFIG_MDIO_BUS_MUX = y : m
CONFIG_CAN_KVASER_USB = m : is not set
CONFIG_SHUFFLE_PAGE_ALLOCATOR = y : is not set
CONFIG_CRYPTO_SHA3_ARM64 = m : is not set
CONFIG_FB_SYS_IMAGEBLIT = y : m
CONFIG_KVM = y : is not set
CONFIG_CRYPTO_CRCT10DIF_ARM64_CE = m : is not set

### CODE REVIEW

File: SPECS/kernel/config [hunk 1]
```diff
@@ -2728,8 +2764,42 @@ CONFIG_STE10XP=m
 # CONFIG_DP83848_PHY is not set
 # CONFIG_DP83867_PHY is not set
 # CONFIG_DP83869_PHY is not set
+# CONFIG_DP83TD510_PHY is not set
 # CONFIG_VITESSE_PHY is not set
 # CONFIG_XILINX_GMII2RGMII is not set
+# CONFIG_PSE_CONTROLLER is not set
+CONFIG_CAN_DEV=m
```
Comment (christopherco) [inline SPECS/kernel/config:2771]: Can we deprecate CAN support?

File: SPECS/kernel/config [hunk 2]
```diff
@@ -3309,8 +3389,13 @@ CONFIG_PINCONF=y
 CONFIG_GENERIC_PINCONF=y
 # CONFIG_DEBUG_PINCTRL is not set
 # CONFIG_PINCTRL_AMD is not set
+# CONFIG_PINCTRL_CY8C95X0 is not set
 # CONFIG_PINCTRL_MCP23S08 is not set
 # CONFIG_PINCTRL_SX150X is not set
+
+#
+# Intel pinctrl drivers
```
Comment (christopherco) [inline SPECS/kernel/config:3397]: Consider deprecating the pinctrl drivers

File: SPECS/kernel/config [hunk 3]
```diff
@@ -4191,51 +4302,147 @@ CONFIG_MEDIA_ATTACH=y
...
+# CONFIG_VIDEO_ST_MIPID02 is not set
+# CONFIG_VIDEO_THS7303 is not set
+# end of Miscellaneous helper chips
 
 CONFIG_MEDIA_TUNER=m
 
 #
 # Customize TV tuners
 #
+CONFIG_MEDIA_TUNER_E4000=m
```
Comment (christopherco) [inline SPECS/kernel/config:4410]: Consider deprecating all these tv tuners

File: SPECS/kernel/config [hunk 4]
```diff
@@ -4245,116 +4452,116 @@ CONFIG_MEDIA_TUNER_QM1D1B0004=m
 #
 # Multistandard (satellite) frontends
 #
+CONFIG_DVB_MXL5XX=m
```
Comment (christopherco) [inline SPECS/kernel/config:4455]: Deprecate these satellite tuners

File: SPECS/kernel/config [hunk 5]
```diff
@@ -4245,116 +4452,116 @@ CONFIG_MEDIA_TUNER_QM1D1B0004=m
...
 CONFIG_DVB_STB0899=m
 CONFIG_DVB_STB6100=m
 CONFIG_DVB_STV090x=m
 CONFIG_DVB_STV0910=m
 CONFIG_DVB_STV6110x=m
 CONFIG_DVB_STV6111=m
-CONFIG_DVB_MXL5XX=m
 
 #
 # Multistandard (cable + terrestrial) frontends
```
Comment (christopherco) [inline SPECS/kernel/config:4464]: 👽

**Review by christopherco — APPROVED** [review]

**Review by Camelron — APPROVED** [review]

**Review by christopherco — APPROVED** [review]

**Review by christopherco — APPROVED** [review]

**Pull request merged**